janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/NumericTranslationQueryParser.java

```diff
@@ -0,0 +1,149 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.diskstorage.lucene;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.document.DoublePoint;
+import org.apache.lucene.document.FloatPoint;
+import org.apache.lucene.document.IntPoint;
+import org.apache.lucene.document.LongPoint;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.Query;
+import org.janusgraph.diskstorage.indexing.KeyInformation;
+import org.janusgraph.graphdb.database.serialize.AttributeUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class NumericTranslationQueryParser extends QueryParser {
+    private static final Logger log = LoggerFactory.getLogger(NumericTranslationQueryParser.class);
+    private final KeyInformation.StoreRetriever storeRetriever;
+
+    public NumericTranslationQueryParser(KeyInformation.StoreRetriever storeRetriever, String field, Analyzer analyzer) {
+        super(field, analyzer);
+        this.storeRetriever = storeRetriever;
+    }
+
+    @Override
+    protected Query newRangeQuery(final String field, final String start, final String end, final boolean startInclusive,
+                                  final boolean endInclusive) {
+
+        Class<?> dataType = storeRetriever.get(field).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return getNumericRangeQuery(field, dataType, start, end, startInclusive, endInclusive);
+            } catch (NumberFormatException e) {
+                log.warn("Expected Number type for " + field + ":" + dataType + ", but " + e.getMessage());
```

Comment:
I am not sure about logging styles but maybe it is better to provide an exception as a separate parameter? I.e. something like:
```
log.warn("Expected Number type for " + field + ":" + dataType + ", but " + " number format was incorrect", e);
```

Reply:
Ok, I move this logger to separated method